### PR TITLE
Add Kotlin port of demo application

### PR DESCRIPTION
**Pull request by cartersande**
This PR adds a Kotlin port of the demo app. The structure of the app is very similar to the Java version, with some small tweaks to make the code easier to understand and take advantage of Kotlin language features.

### CODE REVIEW

**Comment by thomasmso** [comment]
Could you restructure the project, such that your project files don't spill out into the topmost level of the demo app project? The project files are all fairly contained for the original demo app (In the `AppLovin-Android-Demo-App` directory), and would be nice if the same could be done for the Kotlin version as well.

screen shot 2017-06-21 at 2 24 00 pm

**Comment by cartersande** [comment]
@thomasmso These files are generated by Android Studio when you build the project, and they were created even when I tried building a fresh copy of master that didn't have the Kotlin stuff in it. [The branch's tree](https://github.com/AppLovin/Android-SDK-Demo/tree/feat/kotlin_port) doesn't have any new top-level files/folders except a single folder called `AppLovin-Kotlin-Demo-App`.

**Comment by thomasmso** [comment]
Cool, I'll clean things up in a separate branch. Thanks for checking!

File: AppLovin-Kotlin-Demo-App/build.gradle [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    }
+    // Otherwise (AppLovin Demo app is running as a standalone project)
+    else {
+        def libsFolder = new File(projectDir, "libs");
+        if (!libsFolder.exists() || libsFolder.list() == null || libsFolder.listFiles().length == 0) {
+            throw new IllegalStateException("AppLovin SDK jar file was not found in the libs/ folder. Please download the latest AppLovin SDK from http://applovin.com and place it under the libs/ folder of your project")
+        }
+    }
+}
+repositories {
```
Comment (christophercong) [inline AppLovin-Kotlin-Demo-App/build.gradle]: Missing space.

File: AppLovin-Kotlin-Demo-App/src/main/java/com/applovin/apps/kotlindemoapp/banners/BannerLayoutEditorActivity.kt [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+class BannerLayoutEditorActivity : AdStatusActivity()
+{
+
+    override fun onCreate(savedInstanceState: Bundle?)
+    {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_banner_layout_editor)
+
+        adStatusTextView = status_label
```
Comment (christophercong) [inline AppLovin-Kotlin-Demo-App/src/main/java/com/applovin/apps/kotlindemoapp/banners/BannerLayoutEditorActivity.kt]: +1. This is one of favorite uses of kotlin android extension

**Pull request merged**